### PR TITLE
Injecting logger into app/service/docker setup classes

### PR DESCRIPTION
**Pull request by trushton**
Extension of PR #1. The logger object is shared through the multiple classes so that it can be called directly, instead of having a mess of events being fired off and listened for elsewhere. The 'output' event is still used, as it is the one event that all classes were using. 
@kevgo @hugobho @martinjaime @helenmedrano

### CODE REVIEW

File: src/app-setup.ls [hunk 1]
```diff
@@ -9,23 +9,16 @@ require! {
...
 
-  (@app-config) ->
+  (@app-config, @logger) ->
 
 
   start-setup: ->
     setups = for service-name of @app-config.services
-      new ServiceSetup service-name, root: path.join(process.cwd!, @app-config.services[service-name].location)
-        ..on 'start', (name) ~> @emit 'start', name
+      new ServiceSetup service-name, @logger,  root: path.join(process.cwd!, @app-config.services[service-name].location)
```
Comment (kevgo) [inline src/app-setup.ls]: Please make this a named argument, i.e. add it as `logger: @logger` to the second parameter.

File: src/app-setup.ls [hunk 2]
```diff
@@ -9,23 +9,16 @@ require! {
...
-        ..on 'finished', (name, exit-code) ~> @emit 'finished', name, exit-code
-        ..on 'error', (name, exit-code) ~> @emit 'error', name, exit-code
 
     docker-setups = for service-name of @app-config.services
-      new DockerSetup service-name, root: path.join(process.cwd!, @app-config.services[service-name].location)
-        ..on 'docker-exists', (name) ~> @emit 'docker-exists', name
-        ..on 'docker-finished', (name, exit-code) ~> @emit 'docker-finished', name, exit-code
-        ..on 'docker-start', (name) ~> @emit 'docker-start', name
-        ..on 'error', (name, exit-code) ~>   @emit 'docker-error', name, exit-code
+      new DockerSetup service-name, @logger, root: path.join(process.cwd!, @app-config.services[service-name].location)
```
Comment (kevgo) [inline src/app-setup.ls]: same here

File: src/app-setup.ls [hunk 3]
```diff
@@ -9,23 +9,16 @@ require! {
 
 class AppSetup extends EventEmitter
 
-  (@app-config) ->
+  (@app-config, @logger) ->
```
Comment (kevgo) [inline src/app-setup.ls]: There is no intuitive rule why app-config should be argument 1 and logger argument 2. Let's make both arguments named, i.e. `{@app-config, @logger}`.

File: src/docker-setup.ls [hunk 4]
```diff
@@ -27,8 +28,10 @@ class DockerSetup extends EventEmitter
                           stdout: {@write}
                           stderr: {@write})
       ..on 'ended', (exit-code, killed) ~>
-        | exit-code is 0 => @emit 'docker-finished', @name
-        | otherwise      => @emit 'error', @name, exit-code
+        | exit-code is 0 => @logger.log name: @name, text: "Docker setup finished"
+        | otherwise      =>
+            console.log red "\nDocker setup of '#{@name}' failed"
```
Comment (kevgo) [inline src/docker-setup.ls]: Please use the logger for all output. In this case: `logger.error "Docker setup of .... failed"`. The error method should print this stuff in red. If it doesn't exist, feel free to create it.

File: src/service-setup.ls [hunk 5]
```diff
@@ -10,19 +11,21 @@ require! {
...
     new ObservableProcess(call-args(normalize-path @service-config.setup),
                           cwd: @config.root,
                           stdout: {@write}
                           stderr: {@write})
       ..on 'ended', (exit-code, killed) ~>
-        | exit-code is 0 => @emit 'finished', @name
-        | otherwise      => @emit 'error', @name, exit-code
+        | exit-code is 0 => @logger.log name: @name, text: 'setup finished'
+        | otherwise      =>
+            console.log red "\nsetup of '#{@name}' failed with exit code #{exit-code}"
```
Comment (kevgo) [inline src/service-setup.ls]: Same here, use `logger.error`.

File: src/docker-setup.ls [hunk 6]
```diff
@@ -27,8 +28,10 @@ class DockerSetup extends EventEmitter
                           stdout: {@write}
                           stderr: {@write})
       ..on 'ended', (exit-code, killed) ~>
-        | exit-code is 0 => @emit 'docker-finished', @name
-        | otherwise      => @emit 'error', @name, exit-code
+        | exit-code is 0 => @logger.log name: @name, text: "Docker setup finished"
```
Comment (kevgo) [inline src/docker-setup.ls]: 2 spaces around `=>`

**Comment by trushton** [comment]
@kevgo, I had to rebase to master after the other PR was merged in. This PR should also be good for a review

File: src/docker-setup.ls [hunk 7]
```diff
@@ -13,13 +14,13 @@ require! {
...
 
-  (@name, @config) ->
+  ({@name, @logger, @config}) ->
     @service-config = require path.join(@config.root, 'service.yml')
 
 
   start: (done) ~>
-    | @_docker-file-exists!  =>  @emit 'docker-exists', @name; return done!
-    @emit 'docker-start', @name
+    | @_docker-file-exists!  =>  @logger.log name: @name, text "Docker image already exists"; return done!
```
Comment (kevgo) [inline src/docker-setup.ls]: Add a colon after `text`

**Pull request closed without merging**